spec/services/cfe/submit_vehicle_service_spec.rb

```diff
@@ -0,0 +1,48 @@
+require "rails_helper"
+
+RSpec.describe Cfe::SubmitVehicleService do
+  let(:cfe_assessment_id) { SecureRandom.uuid }
+  let(:mock_connection) { instance_double(CfeConnection) }
+
+  describe ".call" do
+    context "when there is vehicle data" do
+      let(:session_data) do
+        {
+          "vehicle_owned" => true,
+          "vehicle_value" => 5556,
+          "vehicle_pcp" => true,
+          "vehicle_finance" => 4445,
+          "vehicle_over_3_years_ago" => true,
+          "vehicle_in_regular_use" => false,
+          "vehicle_in_dispute" => true,
+        }
+      end
+
+      it "calls CFE appropriately" do
+        expect(mock_connection).to receive(:create_vehicles).with(
+          cfe_assessment_id,
+          [{ date_of_purchase: 4.years.ago.to_date,
+             in_regular_use: false,
+             loan_amount_outstanding: 4445,
+             subject_matter_of_dispute: true,
+             value: 5556 }],
+        )
+        described_class.call(mock_connection, cfe_assessment_id, session_data)
+      end
+    end
+
+    context "when the client is asylum supported" do
```

Comment:
should we set a feature flag in this test for asylum and immigration?

Reply:
I actually just reflected on this - I don't think you need to as you're just giving it the right proceeding type so should just 'work' 🤷‍♀️ - I assume!

Reply:
I agree. We actually had a bug last week where a check was started before a feature flag was set and completed after it was set. Part of the CFE logic was making assumptions about whether a flag was _currently_ set, and that broke. So I think we're safer having the services neither know nor care about feature flags if instead they can simply know and care about the contents of the session (and trust that feature flags have determined what gets _into_ the session.